app/lib/actions/katello/content_view_version/incremental_update.rb

```diff
@@ -222,25 +242,31 @@ def copy_deb_content(new_repo, dep_solve, deb_ids)
             unless deb_ids.blank?
               copy_outputs << plan_action(Pulp::Repository::CopyUnits, new_repo.library_instance, new_repo,
                                           ::Katello::Deb.with_identifiers(deb_ids),
-                                        resolve_dependencies: dep_solve).output
+                                        resolve_dependencies: dep_solve,
+                                        incremental: true).output
             end
           end
           copy_outputs
         end
 
-        def copy_yum_content(new_repo, dep_solve, package_ids, errata_ids)
+        def copy_yum_content(new_repo, dep_solve, package_ids, errata_ids, repository_mapping)
+          new_repository_mapping = Hash[repository_mapping.map { |source_repos, dest| [source_repos.map(&:library_instance), dest] }]
```

Comment:
and to further this point, i see: 
``` ::Katello::Repository.generate_yum_dependent_repositories(new_repository_mapping)```

Its not really really clear what this is taking in, nor what its returning.  Some object that holds the mapping and maybe can do transformations on it might be helpful

Reply:
Some further thought, would it be possible to calculate the repository mapping when we need it?  That way we wouldn't have to pass it around all over the place